### ISSUE TITLE
A small question of implementation

### ISSUE THREAD

**Issue by TianQi-777** [IC3Net/IC3Net#7]
Thank you for your sharing, but I have a small question. Why take `prev_hid[0]` and `prev_hid[1]` out of the calculation graph in `trainer.py`. [(click here)](https://github.com/IC3Net/IC3Net/blob/69b7e0ce51a79def593abfef1a976f43e5e13f75/trainer.py#L56).
```
if (t + 1) % self.args.detach_gap == 0:
    if self.args.rnn_type == 'LSTM':
        prev_hid = (prev_hid[0].detach(), prev_hid[1].detach())
    else:
        prev_hid = prev_hid.detach()
```
Looking forward to your reply.

**Comment by tshrjn** [comment]
Hi @TianQi-777 ,

This is because LSTM's have 2 prev states: hidden & cell state.

**Comment by TianQi-777** [comment]
You may misunderstand what I mean, I mean why you use `detach()` to remove these 2 variables (hidden & cell state) from the calculation graph in Pytorch.
